app/context_processors.py

```diff
@@ -22,12 +27,20 @@ def global_context(request):
     if token:
         decoded = jwt.decode(token, secretKey, algorithms=["HS256"])
         userID = decoded.get("userID")
-        username = UsersAPIView.getUser({"id": userID})[0].get("user").get("username")
+
+        data, httpsCode = UsersAPIView.getUsers({"id": userID}, token)
+        if httpsCode != status.HTTP_200_OK:
+            print(f"User update failed: {data.get('message')}")
+            return HttpResponseServerError(f"An error occurred: {data.get('message')}")
+        username = data[0].get("username")
+
     else:
         userID = ""
-        username = ''
+        username = ""
```

Comment:
```suggestion

    userID = ""
    username = ""

    if token:
        decoded = jwt.decode(token, secretKey, algorithms=["HS256"])
        userID = decoded.get("userID")
        username = UsersAPIView.getUser({"id": userID})[0].get("user").get("username")

        data, httpsCode = UsersAPIView.getUsers({"id": userID}, token)
        if httpsCode != status.HTTP_200_OK:
            print(f"User update failed: {data.get('message')}")
            return HttpResponseServerError(f"An error occurred: {data.get('message')}")
            
        username = data[0].get("username")
```

it's good to practice to initially create a variable outside the if/else scope if it's not going to be used within it

Reply:
Good catch, thanks!

Reply:
It won't let commit the suggestion, but I'll make the change here in a moment and push it.